### ISSUE TITLE
How to insert image with custom text?

### ISSUE THREAD

**Issue by lutaev** [michelson/dante2#191]
Hi there.
I'm working on extending of you amazing library. It would be pleasure if you help me with understanding how inserting images working. So, we have in **addButton.js**:
```javascript
insertImage =(file)=> {
  if(!file)
    return

  let opts = {
    url: URL.createObjectURL(file),
  }

  // cleans input image value
  this.fileInput.value = ""

  return this.props.onChange(addNewBlock(this.props.editorState, 'image', opts))
}
```
I just want to add my custom text to be initialized with image, so, I have to add it to `opts` object:

```javascript
let opts = {
  url: URL.createObjectURL(file),
}
```
But if I do it this way, there is no text at all. I also checked **image.js** file and found this string rendering image caption:
```javascript
<EditorBlock {...editorProps}/>
```
How should new `EditorBlock` be initialized to have text?
Thanks.

**Comment by michelson** [comment]
Hi , you could use the `updateTextOfBlock` function located in `
model/index.js`
use it just after insertImage call

note: you have to import the function in the addButton.js file

let me know how it goes

best!


Atte.
Miguel Michelson Martinez
www.vadb.org <http://vadb.org>

**Comment by lutaev** [comment]
Thanks for answer. I already tried `updateTextOfBlock ` but the result was the same - no caption text at all after rendering.

**Comment by michelson** [comment]
Have you called it within the onchange method?

**Comment by lutaev** [comment]
Yes, I did it within the onChange method. Here is full code of method:
```javascript
insertImage =(file)=> {
    if(!file)
      return

    let opts = {
      url: URL.createObjectURL(file),
    }

    // cleans input image value
    this.fileInput.value = ""

    this.props.onChange(addNewBlock(this.props.editorState, 'image', opts))
    setTimeout(() => {
      const { editorState } = this.props
      const currentBlock = getCurrentBlock(editorState)
      this.props.onChange(updateTextOfBlock(editorState, currentBlock, 'My caption'))
    }, 100)
  }
```
I see only image, but not text, editor is empty. According to draftjs, there is no leaves inside editor, possibly this fact will make things clearer.

Thanks again for your patience.

**Comment by michelson** [comment]
Hi @lutaev ,
I've added a branch in which I've implemented the custom caption on image added. As you noted the `updateTextOfBlock` does not work, I've updated the function with the latest api `Modifier`

check it out at, https://github.com/michelson/dante2/tree/default-image-caption

I guess it would be a good starting point

best

**Comment by lutaev** [comment]
Thank you very much. I'll check it.
